File: app/src/main/java/is/xyz/mpv/VersionInfoDialog.java
package is.xyz.mpv;

import android.content.Context;
import android.os.Build;
import android.preference.DialogPreference;
import android.text.method.ScrollingMovementMethod;
import android.util.AttributeSet;
import android.view.View;
import android.widget.TextView;

import org.jetbrains.annotations.NotNull;

public class VersionInfoDialog extends DialogPreference implements LogObserver {
    public VersionInfoDialog(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        init(context);
    }
    public VersionInfoDialog(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context);
    }
    public VersionInfoDialog(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context);
    }

    private String versionText;

    private void init(Context context) {
        setPersistent(false);
        setDialogLayoutResource(R.layout.version_dialog);

        versionText = "mpv-android " + BuildConfig.VERSION_NAME + " / " + BuildConfig.VERSION_CODE + " (" + BuildConfig.BUILD_TYPE + ")\n";
        MPVLib.create(context);
        MPVLib.init();
        MPVLib.addLogObserver(this);
    }

    @Override
    protected void onBindDialogView(View view) {
        super.onBindDialogView(view);

        MPVLib.removeLogObserver(this);
        MPVLib.destroy();
        TextView field = (TextView) view.findViewById(R.id.info);
        field.setText(versionText);
        field.setMovementMethod(new ScrollingMovementMethod());
    }

    @Override
    public void logMessage(@NotNull String prefix, int level, @NotNull String text) {
        if (prefix.equals("cplayer") && level == MPVLib.mpvLogLevel.MPV_LOG_LEVEL_V)
            versionText += text;
    }
}

File: app/src/main/java/is/xyz/mpv/VersionInfoDialog.kt
package `is`.xyz.mpv

import android.content.Context
import android.util.AttributeSet
import android.preference.DialogPreference
import android.text.method.ScrollingMovementMethod
import android.view.View
import android.widget.TextView

class VersionInfoDialog @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = android.R.attr.dialogPreferenceStyle,
    defStyleRes: Int = 0
): DialogPreference(context, attrs, defStyleAttr, defStyleRes), LogObserver {
    private var versionText: String

    init {
        isPersistent = false
        dialogLayoutResource = R.layout.version_dialog

        versionText = "mpv-android ${BuildConfig.VERSION_NAME} / ${BuildConfig.VERSION_CODE} (${BuildConfig.BUILD_TYPE})\n"
        /* create mpv context to capture versions from log */
        MPVLib.create(context)
        MPVLib.init()
        MPVLib.addLogObserver(this)
    }

    override fun onBindDialogView(view: View) {
        super.onBindDialogView(view)

        MPVLib.removeLogObserver(this)
        MPVLib.destroy()
        val field = view.findViewById<TextView>(R.id.info)
        field.text = versionText
        field.movementMethod = ScrollingMovementMethod()
    }

    override fun logMessage(prefix: String, level: Int, text: String) {
        if (prefix == "cplayer" && level == MPVLib.mpvLogLevel.MPV_LOG_LEVEL_V)
            versionText += text
    }
}